source/server/overload_manager_impl.cc

```diff
@@ -374,6 +399,48 @@ bool OverloadManagerImpl::registerForAction(const std::string& action,
   return true;
 }
 
+bool OverloadManagerImpl::tryAllocateResource(OverloadReactiveResourceName resource_name,
```

Comment:
Can you document the return value of this method?  At first glance it seemed to me that the return value should be true when allocation succeeds, but it seems to instead return true if the requested resource exists and this method was able to schedule the provided callback.

Reply:
@antoniovicente at first i was going to proceed with version that returns true if allocation succeeds:
```
bool tryAllocateResource(OverloadReactiveResourceName resource_name, uint64_t increment);
```
But then I could not come up with a way how to call this method from worker thread. I may be wrong, but in my understanding worker thread will not be able to call this method which runs on main thread without either blocking for result or doing it asynchronously (which is done by passing thread dispatcher and a callback).

Reply:
As discussed over slack, this method should be callable from workers.  We just need to figure out how to make it available to workers.